### PR TITLE
Fix Bio.Cluster._cluster doctests

### LINKED ISSUE #2350: Failing doctests in C code
Cross reference #2347 (missing C doctests raising an exception on Python 2), Debian packaging team report issues under Python 3:

https://alioth-lists.debian.net/pipermail/debian-med-packaging/2019-November/074789.html

```
======================================================================
FAIL: kmedoids (Bio.Cluster._cluster)
Doctest: Bio.Cluster._cluster.kmedoids
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python3.8/doctest.py", line 2196, in runTest
    raise self.failureException(self.format_failure(new.getvalue()))
AssertionError: Failed doctest test for Bio.Cluster._cluster.kmedoids
  File "/build/python-biopython-1.75+dfsg/.pybuild/cpython3_3.8/build/Bio/Cluster/_cluster.cpython-38-x86_64-linux-gnu.so", line unknown line number, in kmedoids

----------------------------------------------------------------------
File "/build/python-biopython-1.75+dfsg/.pybuild/cpython3_3.8/build/Bio/Cluster/_cluster.cpython-38-x86_64-linux-gnu.so", line ?, in Bio.Cluster._cluster.kmedoids
Failed example:
    distance = array([[0.0, 1.1, 2.3],
                      [1.1, 0.0, 4.5],
                      [2.3, 4.5, 0.0]])
Exception raised:
    Traceback (most recent call last):
      File "/usr/lib/python3.8/doctest.py", line 1328, in __run
        exec(compile(example.source, filename, "single",
      File "<doctest Bio.Cluster._cluster.kmedoids[0]>", line 1, in <module>
        distance = array([[0.0, 1.1, 2.3],
    NameError: name 'array' is not defined
----------------------------------------------------------------------
File "/build/python-biopython-1.75+dfsg/.pybuild/cpython3_3.8/build/Bio/Cluster/_cluster.cpython-38-x86_64-linux-gnu.so", line ?, in Bio.Cluster._cluster.kmedoids
Failed example:
    distance = array([1.1, 2.3, 4.5])
Exception raised:
    Traceback (most recent call last):
      File "/usr/lib/python3.8/doctest.py", line 1328, in __run
        exec(compile(example.source, filename, "single",
      File "<doctest Bio.Cluster._cluster.kmedoids[1]>", line 1, in <module>
        distance = array([1.1, 2.3, 4.5])
    NameError: name 'array' is not defined
----------------------------------------------------------------------
File "/build/python-biopython-1.75+dfsg/.pybuild/cpython3_3.8/build/Bio/Cluster/_cluster.cpython-38-x86_64-linux-gnu.so", line ?, in Bio.Cluster._cluster.kmedoids
Failed example:
    distance = [array([]),
                array([1.1]),
                array([2.3, 4.5])]
Exception raised:
    Traceback (most recent call last):
      File "/usr/lib/python3.8/doctest.py", line 1328, in __run
        exec(compile(example.source, filename, "single",
      File "<doctest Bio.Cluster._cluster.kmedoids[2]>", line 1, in <module>
        distance = [array([]),
    NameError: name 'array' is not defined


======================================================================
FAIL: pca (Bio.Cluster._cluster)
Doctest: Bio.Cluster._cluster.pca
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python3.8/doctest.py", line 2196, in runTest
    raise self.failureException(self.format_failure(new.getvalue()))
AssertionError: Failed doctest test for Bio.Cluster._cluster.pca
  File "/build/python-biopython-1.75+dfsg/.pybuild/cpython3_3.8/build/Bio/Cluster/_cluster.cpython-38-x86_64-linux-gnu.so", line unknown line number, in pca

----------------------------------------------------------------------
File "/build/python-biopython-1.75+dfsg/.pybuild/cpython3_3.8/build/Bio/Cluster/_cluster.cpython-38-x86_64-linux-gnu.so", line ?, in Bio.Cluster._cluster.pca
Failed example:
    columnmean + dot(coordinates, pc)
Exception raised:
    Traceback (most recent call last):
      File "/usr/lib/python3.8/doctest.py", line 1328, in __run
        exec(compile(example.source, filename, "single",
      File "<doctest Bio.Cluster._cluster.pca[0]>", line 1, in <module>
        columnmean + dot(coordinates, pc)
    NameError: name 'columnmean' is not defined


======================================================================
FAIL: treecluster (Bio.Cluster._cluster)
Doctest: Bio.Cluster._cluster.treecluster
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python3.8/doctest.py", line 2196, in runTest
    raise self.failureException(self.format_failure(new.getvalue()))
AssertionError: Failed doctest test for Bio.Cluster._cluster.treecluster
  File "/build/python-biopython-1.75+dfsg/.pybuild/cpython3_3.8/build/Bio/Cluster/_cluster.cpython-38-x86_64-linux-gnu.so", line unknown line number, in treecluster

----------------------------------------------------------------------
File "/build/python-biopython-1.75+dfsg/.pybuild/cpython3_3.8/build/Bio/Cluster/_cluster.cpython-38-x86_64-linux-gnu.so", line ?, in Bio.Cluster._cluster.treecluster
Failed example:
    distance = array([[0.0, 1.1, 2.3],
                      [1.1, 0.0, 4.5],
                      [2.3, 4.5, 0.0]])
Exception raised:
    Traceback (most recent call last):
      File "/usr/lib/python3.8/doctest.py", line 1328, in __run
        exec(compile(example.source, filename, "single",
      File "<doctest Bio.Cluster._cluster.treecluster[0]>", line 1, in <module>
        distance = array([[0.0, 1.1, 2.3],
    NameError: name 'array' is not defined
----------------------------------------------------------------------
File "/build/python-biopython-1.75+dfsg/.pybuild/cpython3_3.8/build/Bio/Cluster/_cluster.cpython-38-x86_64-linux-gnu.so", line ?, in Bio.Cluster._cluster.treecluster
Failed example:
    distance = array([1.1, 2.3, 4.5])
Exception raised:
    Traceback (most recent call last):
      File "/usr/lib/python3.8/doctest.py", line 1328, in __run
        exec(compile(example.source, filename, "single",
      File "<doctest Bio.Cluster._cluster.treecluster[1]>", line 1, in <module>
        distance = array([1.1, 2.3, 4.5])
    NameError: name 'array' is not defined
----------------------------------------------------------------------
File "/build/python-biopython-1.75+dfsg/.pybuild/cpython3_3.8/build/Bio/Cluster/_cluster.cpython-38-x86_64-linux-gnu.so", line ?, in Bio.Cluster._cluster.treecluster
Failed example:
    distance = [array([]),
                array([1.1]),
                array([2.3, 4.5])]
Exception raised:
    Traceback (most recent call last):
      File "/usr/lib/python3.8/doctest.py", line 1328, in __run
        exec(compile(example.source, filename, "single",
      File "<doctest Bio.Cluster._cluster.treecluster[2]>", line 1, in <module>
        distance = [array([]),
    NameError: name 'array' is not defined


----------------------------------------------------------------------
Ran 521 tests in 210.480 seconds

FAILED (failures = 1)
```

Evidently our continuous integration setup is not looking at the compiled C code modules - but the Debian test environment does.

### PR DESCRIPTION
**Pull request by mr-c**
This pull request fixes #2350 

- [x] I hereby agree to dual licence this and any previous contributions under both
the _Biopython License Agreement_ **AND** the _BSD 3-Clause License_.

- [x] I have read the ``CONTRIBUTING.rst`` file, have run ``flake8`` locally, and
understand that AppVeyor and TravisCI will be used to confirm the Biopython unit
tests and style checks pass with these changes.

- [x] I have added my name to the alphabetical contributors listings in the files
``NEWS.rst`` and ``CONTRIB.rst`` as part of this pull request, am listed
already, or do not wish to be listed. (*This acknowledgement is optional.*)

### CODE REVIEW

File: Bio/Cluster/clustermodule.c [hunk 1]
```diff
@@ -2318,7 +2320,7 @@ static char pca__doc__[] =
 "Adding the column means to the dot product of the coordinates and the\n"
 "principal components, i.e.\n"
 "\n"
-"    >>> columnmean + dot(coordinates, pc)\n"
+"   'columnmean + dot(coordinates, pc)'\n"
```
Comment (mdehoon) [inline Bio/Cluster/clustermodule.c]: Do we need the single quotes here?

Comment (mr-c) [reply Bio/Cluster/clustermodule.c]: I think `>>>` was used to give it nice source code formatting, but that triggered a doctest. So I thought at least a single quote would be better than nothing. Feel free to add a commit that uses that preferred formatting for non-tested code!

Comment (mdehoon) [reply Bio/Cluster/clustermodule.c]: I was just thinking that the quotes make it look like a string. If you remove the single quotes, I am OK with this pull request.

**Comment by mr-c** [comment]
@mdehoon Done, thanks!

**Comment by mr-c** [comment]
Can someone with appveyor permissions hit the "try again" button? Or should I make a trivial change and force push?

**Comment by peterjc** [comment]
Rebuilding...

**Comment by peterjc** [comment]
Thank you both.

**Pull request merged**